spec/models/search/user_search_spec.rb

```diff
@@ -8,21 +10,71 @@
     let!(:first_user) { create(:user_group_assignment, user_group: first_group).user }
     let!(:second_user) { create(:user_group_assignment, user_group: first_group).user }
     let!(:third_user) { create(:user_group_assignment, user_group: second_group).user }
-    subject { User.with_groups.do_search(User.with_groups, %[group:"#{group_sought.name}"]).to_a }
 
-    context "searching for first group" do
-      let(:group_sought) { first_group }
+    # use User.all because rel needs to be an ActiveRecord relation
+    subject { User.do_search(User.all, query, scope).to_a }
+
+    context "searching by group" do
+      let(:query) { %(group:"#{group_sought.name}") }
+      let(:scope) { nil }
+
+      context "searching for first group" do
+        let(:group_sought) { first_group }
+
+        it "should work" do
+          expect(subject).to contain_exactly(first_user, second_user)
+        end
+      end
+
+      context "searching for first group" do
+        let(:group_sought) { second_group }
 
-      it "should work" do
-        expect(subject).to contain_exactly(first_user, second_user)
+        it "should work" do
+          expect(subject).to contain_exactly(third_user)
+        end
       end
     end
 
-    context "searching for first group" do
-      let(:group_sought) { second_group }
+    context "searching by role" do
+      let(:query) { %(role:"#{role_sought}") }
+      let(:other_mission) { create(:mission) }
+
+      before(:each) do
+        first_user.assignments.create!(mission: get_mission, role: "enumerator")
+        first_user.assignments.create!(mission: other_mission, role: "staffer")
+        second_user.assignments.create!(mission: get_mission, role: "coordinator")
+        third_user.assignments.create!(mission: get_mission, role: "staffer")
+      end
+
+      context "in mission" do
+        let(:scope) { {mission: get_mission} }
+
+        context "searching for staffer" do
+          let(:role_sought)  { "staffer" }
+
+          it "should return all users with staffer role in current mission only" do
+            expect(subject).to contain_exactly(third_user)
+          end
+        end
+
+        context "searching for staffer" do
+          let(:role_sought) { "enumerator" }
+
+          it "should return all users with enumerator role in current mission only" do
+            expect(subject).to contain_exactly(first_user)
+          end
```

Comment:
What does this test cover that the first one does not? May be sth but I'm not seeing it. Seems to me you want to test that 1) you don't return any users with the wrong role, and 2) that you only return users that have the role _in the current mission_, not if they have the role in some other mission. I think you're doing both with the first test, no?